### DISCUSSION TITLE
It’s a bug for 2.7.0/2.7.1 use more flash than 2.6.0

### DISCUSSION THREAD

**Discussion by sunnyguhz** [stm32duino/Arduino_Core_STM32#2218] [Q&A] [answered]
When I use 2.7.0/2.7.1 for STM32G030/STM32F1/STM32F0, it's use very more flash than 2.6.0

I made a lot of comparisons, including new blank projects and existing old projects. I don't know what caused this error.
However, through comparison, it was found that 2.5.0 and 2.6.0 do not have this issue.
Only the latest 2.7.0 and 2.7.1 have this issue.
And, Optimize by selecting Smallest (- OS default) and Smallest (- OS) with LTO without any changes.

**Comment by fpistm** [comment]
Could you be more precise. What is the "very more flash?

**Reply by fpistm** [reply]
Note that CMSIS changed from version 5.7.0 to 5.9.0. 
If linked can't do nothing.

**Comment by sunnyguhz** [comment]
For example, creating a new sketch without any program，compiling it.
```
void setup() {
  // put your setup code here, to run once:

}

void loop() {
  // put your main code here, to run repeatedly:

}
```
Debug symbols and core logs:"None"
Optimize:"Smallest (-OS) with LTO"
Board part number:"Generic G030C8Tx"
U(S)ART support:"Disabled (no Serial support)"

When I used 2.6.0, Sketch uses 3144 bytes (4%) of program storage space.  Maximum is 65536 bytes
When I used 2.7.1, Sketch uses 9420 bytes (14%) of program storage space. Maximum is 65536 bytes


My another old Sketch uses 26816 bytes (40%) of program storage space with 2.6.0, and uses 42028 bytes (64%) of program storage space with 2.7.1

**Reply by fpistm** [reply]
Optimize for debug is not selected?

**Reply by fpistm** [reply]
Which ide you used? And version ?

**Comment by sunnyguhz** [comment]
> Which ide you used? And version ?
Arduino IDE 2.2.1,windows10 22H2

**Comment by fpistm** [comment]
Here my test results using freshly installed Arduino IDE 2.2.1.
Blank sketch for genericG030C8Tx. Default options.
## 2.6.0
```
Sketch uses 10280 bytes (15%) of program storage space. Maximum is 65536 bytes.
Global variables use 1180 bytes (14%) of dynamic memory, leaving 7012 bytes for local variables. Maximum is 8192 bytes.
```
## 2.7.1
```
Sketch uses 10328 bytes (15%) of program storage space. Maximum is 65536 bytes.
Global variables use 1180 bytes (14%) of dynamic memory, leaving 7012 bytes for local variables. Maximum is 8192 bytes.
```
So delta is 48 bytes of flash. Don't know how you got those results. Maybe Arduino IDE cache files, Customization you made using *.local.txt or other.
Anyway nothing coming from the core.

**Reply by sunnyguhz** [reply]
Can you set options like this? And try it again
Optimize:"Smallest (-OS) with LTO"
U(S)ART support:"Disabled (no Serial support)"

**Reply by fpistm** [reply]
Exactly the same result with both version 2.6.0/2.7.1 with USART disabled and LTO enabled:
```
Sketch uses 3244 bytes (4%) of program storage space. Maximum is 65536 bytes.
Global variables use 804 bytes (9%) of dynamic memory, leaving 7388 bytes for local variables. Maximum is 8192 bytes.
```

**Reply by fpistm** [reply]
Just a though, check if the  "Optimize for debug" is not checked:
![image](https://github.com/stm32duino/Arduino_Core_STM32/assets/20641798/17d76678-0bb1-4695-bfbc-ad4ef13a0b88)

If it is set then you simply build in -Og -g. 2.7.X added the support of this IDE option and so override the selected optimization. See #2184.
Else Arduino IDE has probably an issue. Maybe the cache.

**Reply by sunnyguhz** [reply, marked as answer]
Yes，"Optimize for debug" is checked. 
And it is right when I uncheck it.

**Reply by fpistm** [reply]
So no issue. :wink:

But next time, answer to the question, I already asked 2 days ago:
https://github.com/orgs/stm32duino/discussions/2218#discussioncomment-7737238
